src/StructuredLogger/Serialization/Serialization.cs

```diff
@@ -46,9 +47,9 @@ public static class Serialization
         public static Build ReadBuildLog(Stream stream, byte[] projectImportsArchive = null) => BuildLogReader.Read(stream, projectImportsArchive);
         public static Build ReadBinLog(Stream stream, byte[] projectImportsArchive = null) => BinaryLog.ReadBuild(stream, projectImportsArchive);
 
-        public static Build Read(string filePath) => Read(filePath, progress: null);
+        public static Build Read(string filePath) => Read(filePath, progress: null, forwardCompatibilitySettings: null);

```

Comment:
I'd pass a simple class ReaderSettings, and on it an enum `UnknownRecordBehavior` `None`, `Error`, `Warning`, `Message`, `ThrowException`

Reply:
Sounds good.
1 message per whole binlog as well sounds enough.
Just not sure what `UnknownRecordBehavior` would mean?
Shouldn't `Error` be a default; `Warning`, `Message`, `None` be opt-in tunning down of the reporting; and `Throw..` be opt-in of the former behavior? Feels like enough
Or did I misunderstood?